superset/db_engine_specs/athena.py

```diff
@@ -64,3 +73,37 @@ def _mutate_label(label: str) -> str:
         :return: Conditionally mutated label
         """
         return label.lower()
+
+    @classmethod
+    def create_table_from_csv(  # pylint: disable=too-many-arguments
```

Comment:
it seems to me that a cleaner solution would be to move chunksize to the config / make it per engine & set it to lower value for athena
https://github.com/apache/superset/blob/790ac5ef870815978f56ce81b806d43c18421940/superset/views/database/views.py#L193

Reply:
Agreed it does seem cleaner and it was my first idea for solving this, but I fear even with a lower chunk size we may still run into problems when trying to upload certain datasets (e.g. CSVs containing long text columns)

pyathena's to_csv converts the dataframe to parquet and uploads it to S3, bypassing this limitation completely

Reply:
I see, in that case I would suggest to follow: https://github.com/apache/superset/blob/efeb5a93e62440ddf16f16aa0bd8aeac52eac4d5/superset/db_engine_specs/hive.py#L191

and implement support for various csv controls like null values, custom separator etc

looks like not all of those supported here: https://github.com/laughingman7743/PyAthena/blob/5942373e31668367ddd2fb9fdd3b9c496d388ec7/pyathena/pandas/util.py#L134

Reply:
Sorry, I'm confused. Aren't these csv controls already handled by `csv_to_df`?
Are you suggesting to move the csv controls to the create table definition and upload the file as it is, like in the example you linked?

Reply:
After having the time to further test the feature I can see this fix clearly needs additional work. Append and Replace are not supported and can corrupt/duplicate your data if used. Will try to do a more proper implementation following @bkyryliuk 's advice.